### ISSUE TITLE
Stop using `object-literal-sort-keys`

### ISSUE THREAD

**Issue by andy-hanson** [palantir/tslint#2583]
This rule is disabled in 133 files! Wouldn't it be easier to just disable it in the config?

**Comment by dgreene1** [comment]
omg please remove this. I can't even get the disabling to work well in vscode. To me, the ordering of imports is personal and up to the dev.

**Comment by vajahath** [comment]
I would suggest to add a `--fix`er for this. 
it would sort it as needed..
well ordered sorted keys in the code looks awesome :stadium: and clean.

**Comment by breathe** [comment]
I can't imagine anyone wanting this rule turned on without an auto-fix for it ...?  Would a feature-request bug to create the auto-fix for object-literals-sort-keys be helpful?

**Comment by adidahiya** [comment]
We generally accept auto-fixer contributions for all rules, so a separate Github issue tracking it is not necessary.

Note that this original ticket was filed to track the enabling of `object-literal-sort-keys` in _this_ source repo, not in TSLint distributions.

**Comment by serhiipalash** [comment]
> We generally accept auto-fixer contributions for all rules

Auto-fixer for `object-literal-sort-keys` could produce very hard to debug errors. Read this https://github.com/prettier/prettier/issues/1096

**Comment by tharakawj** [comment]
I personally don't like this rule because sometimes it's not intuitive to write keys in alphabetical order. As an example if there is a schedule object,
`{ start: '', end: '', frequency: '' } `
order make more sense to me than 
`{ end: '', frequency: '', start: '' }`.
I know it's a personal preference and that's the whole point to remove this rule from recommended preset.

**Comment by mlg87** [comment]
any idea why this config would still have the rule applying and preventing me from compiling?
```json
{
  "extends": ["tslint:recommended", "tslint-react", "tslint-config-prettier"],
  "linterOptions": {
    "exclude": ["config/**/*.js", "node_modules/**/*.ts"]
  },
  "rules": {
    "object-literal-sort-keys": false,
    "no-console": false
  },
  "jsRules": {
    "object-literal-sort-keys": false
  }
}
```

**Comment by cjduncana** [comment]
> I personally don't like this rule because sometimes it's not intuitive to write keys in alphabetical order. As an example if there is a schedule object,
> `{ start: '', end: '', frequency: '' } `
> order make more sense to me than
> `{ end: '', frequency: '', start: '' }`.
> I know it's a personal preference and that's the whole point to remove this rule from recommended preset.

The rationale behind `match-declaration-order` is to implement this order.

**Comment by JoshuaKGoldberg** [comment]
The discussion here has gotten a little off topic 😊 closing for housekeeping purposes.

tl;dr:
* If you don't want the rule, disable it in your `tslint.json` with an `"object-literal-sort-keys": false"` line
* See https://palantir.github.io/tslint/rules/object-literal-sort-keys/ for configuration options that can make it more palatable
